regression_test.go

```diff
@@ -0,0 +1,79 @@
+package veneur
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	proto "github.com/golang/protobuf/proto"
+	"github.com/stripe/veneur/samplers"
+	"github.com/stripe/veneur/ssf"
+)
+
+func validSample() ssf.SSFSpan {
+	return *&ssf.SSFSpan{
+		TraceId:        1,
+		Id:             1,
+		StartTimestamp: 1,
+		EndTimestamp:   10,
+	}
+}
+
+func TestTagNameSet(t *testing.T) {
+	sample := validSample()
+	sample.Tags = make(map[string]string)
+	sample.Tags["name"] = "testName"
+
+	t.Run("nameNotSet", func(t *testing.T) {
+		buf, err := proto.Marshal(&sample)
+		assert.NoError(t, err, "Eror when marshalling sample")
+
+		newSample, metrics, errSSF := samplers.ParseSSF(buf)
+		assert.Equal(t, sample.Tags["name"], newSample.Name, "Name via Tag did not propogate")
+		assert.Zero(t, len(metrics))
+		assert.NoError(t, errSSF)
+	})
+
+	t.Run("nameSet", func(t *testing.T) {
+		sample.Name = "realName"
+
+		buf, err := proto.Marshal(&sample)
+		assert.NoError(t, err, "Error when marshalling sample")
+
+		newSample, metrics, errSSF := samplers.ParseSSF(buf)
+		assert.Equal(t, sample.Name, newSample.Name, "Name did not propogate")
+		assert.Zero(t, len(metrics))
+		assert.NoError(t, errSSF)
+	})
+}
+
+func TestNoTagName(t *testing.T) {
```

Comment:
This is different because `nameSet` still has `sample.Tags["name"] = "testName"` (from the shared test code). I realize now that it's not terribly clear, so I'll separate the tests out.